src/main/java/picard/illumina/CollectIlluminaBasecallingMetrics.java

```diff
@@ -178,22 +196,82 @@ protected int doWork() {
                         IlluminaDataType.PF,
                         IlluminaDataType.Position,
                         IlluminaDataType.Barcodes);
+            }
         }
 
         unmatched_barcode = StringUtil.repeatCharNTimes('N', barcodeLength);
 
         //Initialize data provider, iterate over clusters, and collect statistics
-        final BaseIlluminaDataProvider provider = factory.makeDataProvider();
+        //  BaseIlluminaDataProvider provider;
+        if (USE_NEW_CONVERTER) {
+            setupNewDataProvider(factory);
+        } else {
+            final BaseIlluminaDataProvider provider = factory.makeDataProvider();
 
-        while (provider.hasNext()) {
-            final ClusterData cluster = provider.next();
-            addCluster(cluster);
+            while (provider.hasNext()) {
+                final ClusterData cluster = provider.next();
+                addCluster(cluster);
+            }
         }
 
         onComplete();
         return 0;
     }
 
+    private void setupNewDataProvider(final IlluminaDataProviderFactory factory) {
+        final File laneDir = new File(BASECALLS_DIR, IlluminaFileUtil.longLaneStr(LANE));
+
+        final File[] cycleDirs = IOUtil.getFilesMatchingRegexp(laneDir, IlluminaFileUtil.CYCLE_SUBDIRECTORY_PATTERN);
+
+        //CBCLs
+        final List<File> cbcls = Arrays.stream(cycleDirs)
+                .flatMap(cycleDir -> Arrays.stream(IOUtil.getFilesMatchingRegexp(cycleDir,
+                        "^" + IlluminaFileUtil.longLaneStr(LANE) + "_(\\d{1,5}).cbcl$"))).collect(Collectors.toList());
+
+        if (cbcls.size() == 0) {
+            throw new PicardException("No CBCL files found.");
+        }
+
+        IOUtil.assertFilesAreReadable(cbcls);
+
+        //locs
+        final List<AbstractIlluminaPositionFileReader.PositionInfo> locs = new ArrayList<>();
+        final File locsFile = new File(BASECALLS_DIR.getParentFile(), AbstractIlluminaPositionFileReader.S_LOCS_FILE);
+        IOUtil.assertFileIsReadable(locsFile);
+        try (LocsFileReader locsFileReader = new LocsFileReader(locsFile)) {
+            while (locsFileReader.hasNext()) {
+                locs.add(locsFileReader.next());
+            }
+        }
+
+        //filter
+        final Pattern laneTileRegex = Pattern.compile(ParameterizedFileUtil.escapePeriods(
+                ParameterizedFileUtil.makeLaneTileRegex(".filter", LANE)));
+        final File[] filterFiles = NewIlluminaBasecallsConverter.getTiledFiles(laneDir, laneTileRegex);
+
+        IOUtil.assertFilesAreReadable(Arrays.asList(filterFiles));
+        final Pattern barcodeRegex = Pattern.compile(ParameterizedFileUtil.escapePeriods(
+                ParameterizedFileUtil.makeLaneTileRegex("_barcode.txt", LANE)));
+
+
+        final Map<Integer, File> barcodesFiles = new HashMap<>();
+        IOUtil.assertFilesAreReadable(new ArrayList<>(barcodesFiles.values()));
```

Comment:
`barcodesFiles` is empty here. you could put this assert inside the `if` on line 261.